File: files/200-299/257/CH-257.R
library(tidyverse)
library(readxl)

path = "files/200-299/257/CH-257 Date Calculation.xlsx"
test  = read_excel(path, range = "B2:B14") %>% 
  mutate(Dates = as.Date(Dates))

eom = ceiling_date(ymd(paste(2025, 1:12, 1, sep = "-")), "month") - 1
last_mondays = eom - days((wday(eom) - 2) %% 7) 

all.equal(last_mondays, test$Dates, check.attributes = FALSE)
#> [1] TRUE

File: files/200-299/257/CH-257.py
import pandas as pd
path = "200-299/257/CH-257 Date Calculation.xlsx"
test = pd.read_excel(path, usecols="B", skiprows=1, nrows=12)

result = pd.DataFrame({
    "Result": [
        (d - pd.Timedelta(days=d.weekday() % 7)).to_datetime64()
        for d in pd.date_range("2025-01-31", "2025-12-31", freq="ME")
    ]
})

print(result["Result"].equals(test["Dates"])) # True
